File: surf-cloud-core/src/main/java/dev/slne/surf/cloud/core/netty/protocol/packets/server/KeepAlivePacket.java
package dev.slne.surf.cloud.core.netty.protocol.packets.server;

import dev.slne.surf.cloud.api.meta.SurfNettyPacket;
import dev.slne.surf.cloud.api.meta.SurfNettyPacket.DefaultIds;
import dev.slne.surf.cloud.api.netty.packet.NettyPacket;
import dev.slne.surf.cloud.api.netty.protocol.buffer.SurfByteBuf;

@SurfNettyPacket(id = DefaultIds.KEEP_ALIVE_PACKET)
public class KeepAlivePacket extends NettyPacket<KeepAlivePacket> {

  private long id;

  @Override
  public KeepAlivePacket decode(SurfByteBuf buffer) {
  }

  @Override
  public void encode(SurfByteBuf buffer) {

  }
}

File: surf-cloud-core/src/main/kotlin/dev/slne/surf/cloud/core/netty/protocol/packets/server/KeepAlivePacket.kt
package dev.slne.surf.cloud.core.netty.protocol.packets.server

import dev.slne.surf.cloud.api.meta.DefaultIds
import dev.slne.surf.cloud.api.meta.SurfNettyPacket
import dev.slne.surf.cloud.api.netty.packet.NettyPacket
import dev.slne.surf.cloud.api.netty.protocol.buffer.SurfByteBuf
import dev.slne.surf.cloud.core.util.random

@SurfNettyPacket(id = DefaultIds.KEEP_ALIVE_PACKET)
class KeepAlivePacket : NettyPacket<KeepAlivePacket>() {
    var keepAliveId: Long
        private set

    init {
        keepAliveId = random.nextLong()
    }

    override fun decode(buffer: SurfByteBuf): KeepAlivePacket {
        return this
    }

    override fun encode(buffer: SurfByteBuf) {
        buffer.writeLong(keepAliveId)
    }
}
